agent/xds/server.go

```diff
@@ -209,13 +232,13 @@ func (s *Server) process(stream ADSStream, reqCh <-chan *envoy_discovery_v3.Disc
 	// Configure handlers for each type of request
 	handlers := map[string]*xDSType{
 		EndpointType: {
+			generator: g,
 			typeURL:   EndpointType,
-			resources: s.endpointsFromSnapshot,
 			stream:    stream,
 		},
 		ClusterType: {
+			generator: g,
 			typeURL:   ClusterType,
-			resources: s.clustersFromSnapshot,
```

Comment:
This type-switched method call is now handled by a helper function and shared between sotw and delta.